src/theme/mod.rs

```diff
@@ -205,16 +207,86 @@ where
 
 #[derive(PartialEq, Debug, Default)]
 struct ExtensionMappings {
-    mappings: Vec<(glob::Pattern, Style)>,
+    mappings: Vec<GlobPattern>,
+}
+
+#[derive(PartialEq, Debug)]
+enum GlobPattern {
+    Complex(glob::Pattern, Style),
+
+    // TODO: the String here should probably be an inline string, since we're likely working with
+    // small strings (file extensions).
+    Simple(HashMap<String, Style>),
```

Comment:
I don’t think it’s worth it to try small strings. The time to build the hashmap is negligible, and when searching in it hashing will be used so it doesn’t matter if it’s a String as a key. Feel free to try if you really want to, though.